### ISSUE TITLE
SingleActivator doesn't recognize shortcuts with  some 'alt  + key' combinations

### ISSUE THREAD

**Issue by franciscojma86** [flutter/flutter#119088]
## Use case

A SingleActivator created as:
```
SingleActivator(
   LogicalKeyboardKey.keyG,
   meta: true,
   alt: true,
)
```
doesn't work. Because `alt + G` produces `©`,  and SingleActivator uses `RawKeyEvent` under the hood, the activator expects the logical key for `©`, not `G`. 

A workaround for this, would be to use the `CharacterActivator` as:
```
CharacterActivator(
   '©',
   meta: true,
   alt: true,
)
```

but this doesn't cover the case where you want to use the actual G key.

## Proposal

Create a new type of activator that uses the new `KeyEvent` class, which will do the right thing.

**Comment by darshankawar** [comment]
Thanks for the report @franciscojma86 
In [this](https://github.com/flutter/flutter/issues/116658#issuecomment-1344593376) comment, it is suggested to use `CharacterActivator` since `SingleActivator` is not keyboard layout agnostic. Can you check if the workaround you mentioned could be acceptable and also based on the linked comment ?

**Comment by franciscojma86** [comment]
It does! But I believe there is still need for an activator that doesn't get fooled by the generated character. If alt + g didn't generate © in a keyboard, the shortcut wouldn't work.  @gspencergoog @dkwingsmt

**Comment by gspencergoog** [comment]
Yes, @franciscojma86 is right.  For cases involving the option (alt) on macOS, it also means that those shortcuts aren't transportable across platforms: it you wanted Alt+G to work on Windows and Option+G to work on macOS, you currently have to define two different activators: one that is a `CharacterActivator` and one that is a `SingleActivator`.
